src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -76,140 +82,142 @@
+                .flatMap(svg -> Mono.just(ResponseEntity.ok().contentType(MediaType.APPLICATION_XML).body(svg)))
+                .onErrorResume(e -> Mono.just(ResponseEntity.notFound().build()));
     }
 
     @GetMapping(value = "/studies/{studyName}/network/voltage-levels/{voltageLevelId}/svg-and-metadata")
     @ApiOperation(value = "get the voltage level diagram for the given network and voltage level", produces = "application/json")
     @ApiResponse(code = 200, message = "The svg and metadata")
-    public ResponseEntity<String> getVoltageLevelDiagramAndMetadata(
+    public Mono<ResponseEntity<String>> getVoltageLevelDiagramAndMetadata(
             @PathVariable("studyName") String studyName,
             @PathVariable("voltageLevelId") String voltageLevelId,
             @ApiParam(value = "useName") @RequestParam(name = "useName", defaultValue = "false") boolean useName,
             @ApiParam(value = "centerLabel") @RequestParam(name = "centerLabel", defaultValue = "false") boolean centerLabel,
             @ApiParam(value = "diagonalLabel") @RequestParam(name = "diagonalLabel", defaultValue = "false") boolean diagonalLabel,
             @ApiParam(value = "topologicalColoring") @RequestParam(name = "topologicalColoring", defaultValue = "false") boolean topologicalColoring) {
-        UUID networkUuid = studyService.getStudyUuid(studyName);
-        String svgAndMetadata = studyService.getVoltageLevelSvgAndMetadata(networkUuid, voltageLevelId, useName, centerLabel, diagonalLabel, topologicalColoring);
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(svgAndMetadata);
+        return studyService.getStudyUuid(studyName)
+                .flatMap(uuid -> studyService.getVoltageLevelSvgAndMetadata(uuid, voltageLevelId, useName, centerLabel, diagonalLabel, topologicalColoring))
+                .flatMap(svgAndMetadata -> Mono.just(ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(svgAndMetadata)))
+                .onErrorResume(e -> Mono.just(ResponseEntity.notFound().build()));
     }
 
     @GetMapping(value = "/studies/{studyName}/network/voltage-levels")
     @ApiOperation(value = "get the voltage levels for a given network")
     @ApiResponse(code = 200, message = "The voltage level list of the network")
-    public ResponseEntity<List<VoltageLevelAttributes>> getNetworkVoltyutageLevels(@PathVariable("studyName") String studyName) {
-        UUID networkUuid = studyService.getStudyUuid(studyName);
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getNetworkVoltageLevels(networkUuid));
+    public ResponseEntity<List<VoltageLevelAttributes>> getNetworkVoltageLevels(@PathVariable("studyName") String studyName) {
+        Mono<UUID> networkUuid = studyService.getStudyUuid(studyName);
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getNetworkVoltageLevels(networkUuid.block()));
```

Comment:
You needed to block here?

Reply:
No, that will be removed in the next commit.